soroban-sdk/src/num.rs

```diff
@@ -217,7 +217,6 @@ impl U256 {
     }
 
     pub fn from_be_bytes(env: &Env, bytes: &Bytes) -> Self {
-        env.check_same_env(bytes.env()).unwrap_infallible();
```

Comment:
I'm curious why it's fine to remove this.

Reply:
Here is the discussion: https://github.com/stellar/rs-soroban-sdk/pull/1467#discussion_r2094086795

TL;DR I think it's ok to remove this for the time being as it was a semi-arbitrary place for the check - every host function call is subject to the same kind of issue. We could come back to this later to make a more holistic and intentional change.